PWGDQ/dielectron/LMEE/AliAnalysisTaskLMeeCocktailMC.cxx

```diff
@@ -610,20 +837,31 @@ void AliAnalysisTaskLMeeCocktailMC::ProcessMCParticles(){
         feeorigeta=ee.Eta();
         feeorigphi=ee.Phi();
         if(particle->GetPdgCode()>0) { feeorigphiv=PhiV(dau1,dau2); }else{ feeorigphiv=PhiV(dau2,dau1);};
+
         //get the efficiency weight
         Int_t effbin=fhwEffpT->FindBin(dau1.Pt());
         fwEffpT=fhwEffpT->GetBinContent(effbin);
         effbin=fhwEffpT->FindBin(dau2.Pt());
         fwEffpT=fwEffpT*fhwEffpT->GetBinContent(effbin);
 
         //Resolution and acceptance  
-        dau1=ApplyResolution(dau1);
-        dau2=ApplyResolution(dau2);
+        //-------------------------
+        if(particle->GetPdgCode()>0) { dau1=ApplyResolution(dau1,-1,fResolType); }else{ dau1=ApplyResolution(dau1,1,fResolType);};
+        if(particle2->GetPdgCode()>0) { dau2=ApplyResolution(dau2,-1,fResolType); }else{ dau2=ApplyResolution(dau2,1,fResolType);};
         fpass=kTRUE;
         if(dau1.Pt()<fMinPt||dau2.Pt()<fMinPt) fpass=kFALSE; //leg pT cut
-        if(TMath::Abs(dau1.Rapidity())>fMaxY||TMath::Abs(dau2.Rapidity())>fMaxY) fpass=kFALSE;
+        if(TMath::Abs(dau1.Eta())>fMaxEta||TMath::Abs(dau2.Eta())>fMaxEta) fpass=kFALSE;
 
-        //Fill words after resolution/acceptance
+        //get the pair DCA (based in smeared pT)
+        Float_t DCAtemplateLowEdge[nbDCAtemplate] = {0., .3, .4, .6, 1., 2. };
+        Float_t DCAtemplateUpEdge[nbDCAtemplate]  =     {.3, .4, .6, 1., 2., 100000.}  ;
```

Comment:
Hi @otonvd ,
This seems to be causing an error with clang:

| /alice/sw/SOURCES/AliPhysics/root6/0/PWGDQ/dielectron/LMEE/AliAnalysisTaskLMeeCocktailMC.cxx:856:36: error: variable-sized object may not be initialized
|        Float_t DCAtemplateLowEdge[nbDCAtemplate] = {0., .3, .4, .6, 1., 2. };
|                                   ^~~~~~~~~~~~~
|/alice/sw/SOURCES/AliPhysics/root6/0/PWGDQ/dielectron/LMEE/AliAnalysisTaskLMeeCocktailMC.cxx:857:35: error: variable-sized object may not be initialized
|        Float_t DCAtemplateUpEdge[nbDCAtemplate]  =     {.3, .4, .6, 1., 2., 100000.}  ;
|                                  ^~~~~~~~~~~~~

Could you please fix it?